### ISSUE TITLE
raw wiki files return 404

### ISSUE THREAD

**Issue by TankTheFrank** [go-gitea/gitea#3523]
- Gitea version (or commit ref): b29c3583156ad51d57e24c62eb540290410e9608
- Git version: not relevant
- Operating system: not relevant
- Database (use `[x]`): not relevant
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
   https://try.gitea.io/testerxxx/test-wiki/wiki/raw/file.txt
- Log gist:

## Description

Raw files need to have '.md' added as extension to be served.

They are served by repo.WikiRaw() which does this:

```
...
wikiPath := models.WikiNameToFilename(providedPath)
...
```
models.WikiNameToFilename is a short function:
```
// WikiNameToFilename converts a wiki name to its corresponding filename.
func WikiNameToFilename(name string) string {
	name = strings.Replace(name, " ", "-", -1)
	return url.QueryEscape(name) + ".md"
}
```

Shortly put ".md" is appended to the path, in order to get a raw file from the repo, say "file.txt" it has to be saved as "file.txt.md" and it will be available as "wiki/raw/file.txt".

**Comment by thehowl** [comment]
Seems to work fine here:

https://try.gitea.io/testerxxx/test-wiki/wiki/raw/Home.md

https://try.gitea.io/testerxxx/test-wiki/wiki/raw/Home

**Comment by TankTheFrank** [comment]
"raw files" means non-md files, files that are not markdown and not supposed to be parsed like images, text, etc.

**Comment by Nabile-Rahmani** [comment]
@thehowl Gitea pre-1.4.0 (1.3.2) used to be able to serve any file from a project's wiki, even in subdirectories.
I use wikis in conjunction with mkdocs, and linking to a md source file for editing, but also for storing other documentation file types.

Even though previous versions didn't show the website with the markdown editor, it did serve md files and any other kind of file type present in the wiki repository.
Optionally, it would also be nice if `wiki/_pages` would recursively search for markdown files, but I don't have a need for it.

I can link you an example if you need it.

**Comment by thehowl** [comment]
@Nabile-Rahmani sorry about the delay. Hmm, if it did use to work, this seems like an issue then. @lafriks can you please add kind/bug?

**Comment by Pablo2048** [comment]
Does this bug have something in common with no images shown in wiki pages please? I'm unable to put any image into wiki pages - it always shows 404 error...

**Comment by Pablo2048** [comment]
@thehowl Thanks for reply - No - it seems like external images works. Internal images (contained inside wiki) return 404.

**Comment by thehowl** [comment]
Then yes, it's the same issue. Thanks for pitching in!

**Comment by lunny** [comment]
Is this still a problem?

**Comment by Pablo2048** [comment]
Yes - in 1.7.1 the problem persist. It seems like it will be solved in 1.8 - see https://github.com/go-gitea/gitea/issues/4690

**Comment by Nabile-Rahmani** [comment]
Could someone reopen this issue ? I'm still getting 404 errors on 1.9.4.
